lib/editor/components/pattern/EditSettings.js

```diff
@@ -75,6 +74,21 @@ export default class EditSettings extends Component<Props, State> {
     const noSegmentIsActive = !patternSegment && patternSegment !== 0
     return (
       <div>
+        <ControlLabel><small>Snap to options</small></ControlLabel>
+        <FormControl
+          componentClass='select'
+          value={snapToOption}
+          name={'snapToOption'}
+          onChange={this._onSelectChange}>
+          {SNAP_TO_OPTIONS.map(v => {
+            return (
+              <option
+                key={v}
+                value={v}>{toSentenceCase(v.replace(/_/g, ' '))}
```

Comment:
You could avoid this by using an enum and then having a map from enum to string

Reply:
with JS you could even have one large object that has both enum value and string!